syn/python/get_kge.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import re
+import argparse
+
+weighted_dict = {}
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    cell_name = ''
+    for line in lib_file:
+        if re.search(r'^  cell \(', line):
+            if cell_name != '':
+                print("Size missing")
+                quit()
```

Comment:
This would probably be cleaner using exceptions (`raise RuntimeError('size missing')`) or with a `die_horribly` function, which takes a message to write to `sys.stderr` and then exits with code 1. For bonus points, a line number might be helpful. The easiest way to do that is to change line 21 to `for line_idx, line in enumerate(lib_file):` (and remember that it starts at zero, so error messages need to talk about line `line_idx + 1`).

Reply:
Nice - thanks for the suggestion, I'll have a go at that